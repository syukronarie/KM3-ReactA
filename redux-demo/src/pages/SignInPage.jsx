import React from "react";
import { Link, useNavigate } from "react-router-dom";
import APIAuth from "../apis/auth.api";

const SignInPage = () => {
	const navigate = useNavigate();

	const handleSubmit = async (e) => {
		e.preventDefault();
		const formData = new FormData(e.target);
		const email = formData.get("email");
		const password = formData.get("password");
		await APIAuth.signin({ email, password })
			.then((result) => {
				result.data.success && navigate("/");
			})
			.catch(() => {
				alert("password or email is wrong");
			});
	};

	return (
		<div>
			<form onSubmit={handleSubmit} className="form-container">
				<label htmlFor="email">Email</label>
				<input type="email" name="email" id="email" />
				<label htmlFor="password">Password</label>
				<input type="password" name="password" id="password" />
				<button type="submit" data-testid="button-submit">
					Sign In
				</button>
				<Link to="/register">Register</Link>
			</form>
		</div>
	);
};

export default SignInPage;
