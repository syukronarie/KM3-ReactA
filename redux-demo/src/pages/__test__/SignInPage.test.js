import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import SignInPage from "../SignInPage";

describe("SignInPage Testing", () => {
	it("SignInPage Snapshot", () => {
		const { container } = render(<SignInPage />);
		expect(container).toMatchSnapshot();
	});

	it("should hit handle submit correctly", () => {
		render(<SignInPage />);
		const submitButton = screen.getByTestId("button-submit");
		userEvent.click(submitButton);
	});

	it("should render submit button as Sign In", () => {
		render(<SignInPage />);
		const submitButton = screen.getByTestId("button-submit");
		expect(submitButton).toHaveTextContent(/sign in/i);
	});
});
