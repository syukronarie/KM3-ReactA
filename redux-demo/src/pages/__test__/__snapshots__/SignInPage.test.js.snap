// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInPage Testing 1`] = `
<div>
  <div>
    <form
      class="form-container"
    >
      <label
        for="email"
      >
        Email
      </label>
      <input
        id="email"
        name="email"
        type="email"
      />
      <label
        for="password"
      >
        Password
      </label>
      <input
        id="password"
        name="password"
        type="password"
      />
      <button
        type="submit"
      >
        Sign In
      </button>
    </form>
  </div>
</div>
`;

exports[`SignInPage Testing SignInPage Snapshot 1`] = `
<div>
  <div>
    <form
      class="form-container"
    >
      <label
        for="email"
      >
        Email
      </label>
      <input
        id="email"
        name="email"
        type="email"
      />
      <label
        for="password"
      >
        Password
      </label>
      <input
        id="password"
        name="password"
        type="password"
      />
      <button
        data-testid="button-submit"
        type="submit"
      >
        Sign In
      </button>
    </form>
  </div>
</div>
`;
